lib/msf/core/payload/windows/stageless_meterpreter.rb

```diff
@@ -75,11 +76,8 @@ def generate_stageless_meterpreter(url = nil)
 
     # the URL might not be given, as it might be patched in some other way
     if url
-      url = "s#{url}\x00"
-      location = dll.index("https://#{'X' * 256}")
-      if location
-        dll[location, url.length] = url
-      end
+      # Patch the URL using the patcher as this upports both ASCII and WCHAR.
+      Rex::Payloads::Meterpreter::Patch.patch_string!(dll, "https://#{'X' * 512}", "s#{url}\x00")
```

Comment:
When the string is patched as wchar, it ends up as a double-null. It's left as a single NULL in cases where the payload is still ASCII.